sam.c

```diff
@@ -352,6 +352,33 @@ int32_t bam_endpos(const bam1_t *b)
         return b->core.pos + 1;
 }
 
+static void bam_tag2cigar(bam1_t *b)
+{
+    bam1_core_t *c = &b->core;
+    uint32_t cigar_st, n_cigar4, CG_st, CG_en, ori_len = b->l_data;
+    uint8_t *CG;
+    if (c->n_cigar > 0 || !(c->flag & BAM_FUNMAP) || c->tid < 0 || c->pos < 0) return;
+    if ((CG = bam_aux_get(b, "CG")) == 0) return;
+    if (CG[0] != 'B' || CG[1] != 'I') return;
+    cigar_st = (uint8_t*)bam_get_cigar(b) - b->data;
+    c->n_cigar = *(uint32_t*)(CG + 2);
+    n_cigar4 = c->n_cigar * 4;
+    CG_st = CG - b->data - 2;
+    CG_en = CG_st + 8 + n_cigar4;
+    b->l_data += n_cigar4;
+    if (b->m_data < b->l_data) {
+        b->m_data = b->l_data;
+        kroundup32(b->m_data);
+        b->data = (uint8_t*)realloc(b->data, b->m_data);
+    }
```

Comment:
I take it all back. :-)

The ```bam_aux_get``` call already checks (since 95b1034e93b345fef7aa4980750ba1dedde2c284 by Rob I think) the data being queried doesn't overflow the buffer and returns NULL if so, making the entire ```bam_tag2cigar``` function bail out at this point.  Alarm averted! :-)